synapse/models/inet.py

```diff
@@ -602,31 +249,404 @@ def norm(self, valu, oldval=None):
             subs['port'] = port
 
         if resauth:
-            hostpart = '%s@%s' % (resauth,hostpart)
+            hostpart = '%s@%s' % (resauth, hostpart)
 
-        valu = '%s://%s/%s' % (proto,hostpart,respath)
-        return valu,subs
+        valu = '%s://%s/%s' % (proto, hostpart, respath)
+        return valu, subs
 
     def repr(self, valu):
         return valu
 
-
 class CidrType(DataType):
-
     def norm(self, valu, oldval=None):
-
-        ipstr,maskstr = valu.split('/')
+        ipstr, maskstr = valu.split('/')
 
         mask = int(maskstr)
         ipv4 = ipv4int(ipstr)
 
         if mask > 32 or mask < 0:
             self._raiseBadValu(valu, mesg='Invalid CIDR Mask')
 
-        ipv4 = ipv4mask(ipv4,mask)
-        valu = '%s/%d' % ( ipv4str(ipv4), mask )
+        ipv4 = ipv4mask(ipv4, mask)
+        valu = '%s/%d' % (ipv4str(ipv4), mask)
 
-        return valu,{'ipv4':ipv4,'mask':mask}
+        return valu, {'ipv4': ipv4, 'mask': mask}
 
     def repr(self, valu):
         return valu
+
+class InetMod(CoreModule):
+
+    def initCoreModule(self):
+        # add an inet:defang cast to swap [.] to .
+        self.core.addTypeCast('inet:defang', castInetDeFang)
+        self.onFormNode('inet:fqdn', self.onTufoFormFqdn)
+        self.onFormNode('inet:passwd', self.onTufoFormPasswd)
```

Comment:
Did onFormNode end up feeling helpful, or more like a contrivance?  I wasn't sure if it was ultimately better to just have developers use @on() decorators for node formation, especially if we ever end up needing to plumb more junk into the extensible event info...  Any thoughts?